drivers/SmartThings/zigbee-switch/src/init.lua

```diff
@@ -85,6 +100,133 @@ local device_init = function(self, device)
   end
 end
 
+
+local function send_discover_command(device)
+  local discover_attribute_id = 0x0c
+  local discover_command_payload = "\x00\x00\xff"      -- Start Attribute: 0x0000, Maximum Number: 255
+  local header_args = {
+    cmd = data_types.ZCLCommandId(discover_attribute_id)
+  }
+  local zclh = zcl_messages.ZclHeader(header_args)
+  local addrh = messages.AddressHeader(
+    zb_const.HUB.ADDR,
+    zb_const.HUB.ENDPOINT,
+    device:get_short_address(),
+    device:get_endpoint(CLUSTER_BASIC),
+    zb_const.HA_PROFILE_ID,
+    CLUSTER_BASIC
+  )
+  local message_body = zcl_messages.ZclMessageBody({
+    zcl_header = zclh,
+    zcl_body = generic_body.GenericBody(discover_command_payload)
+  })
+  local send_message = messages.ZigbeeMessageTx({
+      address_header = addrh,
+      body = message_body
+  })
+  device:send(send_message)
+end
+
+local function read_attribute_function(device, cluster_id, attr_id)
+  local read_body = read_attribute.ReadAttribute( attr_id )
+  local zclh = zcl_messages.ZclHeader({
+    cmd = data_types.ZCLCommandId(read_attribute.ReadAttribute.ID)
+  })
+
+  local addrh = messages.AddressHeader(
+      zb_const.HUB.ADDR,
+      zb_const.HUB.ENDPOINT,
+      device:get_short_address(),
+      device:get_endpoint(cluster_id.value),
+      zb_const.HA_PROFILE_ID,
+      cluster_id.value
+  )
+  local message_body = zcl_messages.ZclMessageBody({
+    zcl_header = zclh,
+    zcl_body = read_body
+  })
+  return messages.ZigbeeMessageTx({
+    address_header = addrh,
+    body = message_body
+})
+end
+
+local function device_added(driver, device, event)
+  device:set_find_child(find_child)
+  
+  local num_switch_server_eps = 0
+  local main_endpoint = device:get_endpoint(clusters.OnOff.ID)
+  local updated_flag = false
+  
+  send_discover_command(device)
+  for _, ep in ipairs(device.zigbee_endpoints) do
+   
+    num_switch_server_eps = num_switch_server_eps + 1
+    if ep.id ~= main_endpoint then 
+      if updated_flag == false then
+        device:try_update_metadata({profile="basic-switch"})
+        updated_flag = true
+      end
+      if device:supports_server_cluster(clusters.OnOff.ID, ep.id) then
+        if find_child(device, num_switch_server_eps) == nil then
+          local name = string.format("%s %d", device.label, num_switch_server_eps)
+          local child_profile = "basic-switch"
+          driver:try_create_device(
+            {
+              type = "EDGE_CHILD",
+              label = name,
+              profile = child_profile,
+              parent_device_id = device.id,
+              parent_assigned_child_key = string.format("%02X", num_switch_server_eps),
+              vendor_provided_label = name
+            }
+          )
```

Comment:
```suggestion
          driver:try_create_device(
            {
              type = "EDGE_CHILD",
              label = name,
              profile = "basic-switch",
              parent_device_id = device.id,
              parent_assigned_child_key = string.format("%02X", num_switch_server_eps),
              vendor_provided_label = string.format("%s %d", device.label, num_switch_server_eps)
            }
          )
```
simple and easy to understand